data-products/src/main/scala/org/sunbird/analytics/audit/CourseBatchStatusUpdaterJob.scala

```diff
@@ -0,0 +1,158 @@
+    JobLogger.log(s"Course Batch status metrics before updating the table", Option(metrics), INFO)
+    val res = updateBatchStatus(config, collectionBatchDF)
+    collectionBatchDF.unpersist()
+    res
+  }
+
+  def updateBatchStatus(updaterConfig: JobConfig, collectionBatchDF: DataFrame)(implicit sc: SparkContext): CourseBatchStatusMetrics = {
+    val dateFormatter = new SimpleDateFormat("yyyy-MM-dd")
+    dateFormatter.setTimeZone(TimeZone.getTimeZone("IST"))
+    val currentDate = dateFormatter.format(new Date)
+
+    val finalDF = collectionBatchDF.withColumn("updated_status",
+      when(lit(currentDate).gt(col("enddate")), 2)
+        .otherwise(
+          when(lit(currentDate).geq(col("startdate")), 1).otherwise(col("status"))
+        ))
+      .filter(col("updated_status") =!= col("status"))
+      .drop("status").withColumnRenamed("updated_status", "status")
+    JobLogger.log(s"Writing records into database", None, INFO)
+    finalDF.write.format("org.apache.spark.sql.cassandra").options(collectionBatchDBSettings ++ Map("confirm.truncate" -> "false")).mode(SaveMode.Append).save()
+    if (!finalDF.isEmpty) {
+      updateCourseBatchES(finalDF.filter(col("status") > 0).select("batchid", "status").collect.map(r => Map(finalDF.select("batchid", "status").columns.zip(r.toSeq): _*)), updaterConfig)
+      updateCourseMetadata(finalDF.select("courseid").collect().map(_ (0)).toList.asInstanceOf[List[String]].distinct, finalDF.filter(col("status") < 2), dateFormatter, updaterConfig)
+    } else {
+      JobLogger.log("No records found to update the db", None, INFO)
+    }
+    CourseBatchStatusMetrics(finalDF.filter(col("status") === 0).count(), finalDF.filter(col("status") === 1).count(), finalDF.filter(col("status") === 2).count())
+  }
+
+  def getCollectionBatchDF(fetchData: (SparkSession, Map[String, String], String, StructType) => DataFrame)(implicit spark: SparkSession): DataFrame = {
+    fetchData(spark, collectionBatchDBSettings, cassandraFormat, new StructType()).select("courseid", "batchid", "startdate", "name", "enddate", "enrollmentenddate", "enrollmenttype", "createdfor", "status")
+  }
+
+  def getCourseMetaData(row: Row, dateFormat: SimpleDateFormat): Map[String, AnyRef] = {
+    Map[String, AnyRef](
+      "batchId" -> row.getAs[String]("batchid"),
+      "startDate" -> row.getAs[String]("startdate"),
+      "enrollmentType" -> row.getAs[String]("enrollmenttype"),
+      "createdFor" -> row.getAs[List[String]]("createdfor"),
+      "status" -> row.getAs[AnyRef]("status"),
+      "enrollmentEndDate" -> getEnrolmentEndDate(row.getAs[String]("enrollmentenddate"), row.getAs[String]("enddate"), dateFormat)
+    )
+  }
+
+  def getEnrolmentEndDate(enrolmentDate: String, enDate: String, dateFormatter: SimpleDateFormat): String = {
+    Option(enrolmentDate).map(x => x).getOrElse({
```

Comment:
here endDate can also be null.
if enrolmentEndDate is not null -> return enrolmentEndDate
If (end Date is not null) && (enrolmentEndDate is null) -> set previous day of endDate as enrolmentEndDate.
if (endDate is null) && (enrolmentEndDate is null)  -> return null